app/controllers/katello/api/v2/systems_controller.rb

```diff
@@ -20,7 +21,7 @@ class Api::V2::SystemsController < Api::V2::ApiController
 
     before_filter :find_environment_and_content_view, :only => [:create]
     before_filter :find_content_view, :only => [:create, :update]
-    before_filter :load_search_service, :only => [:index, :available_host_collections, :tasks]
+    before_filter :load_search_service, :only => [:index, :available_host_collections, :tasks, :auto_complete_search]
```

Comment:
We shouldn't need to load the search service here if we are switching to scoped search.